app/presenters/account_show_presenter.rb

```diff
@@ -34,10 +36,35 @@ def show_gpo_partial?
     user.gpo_verification_pending_profile?
   end
 
+  def show_ipp_partial?
+    user.pending_in_person_enrollment.present?
+  end
+
   def showing_any_partials?
+    show_password_reset_partial? ||
+      show_gpo_partial? || show_ipp_partial? || show_pii_partial?
+  end
+
+  def showing_alerts?
     show_service_provider_continue_partial? ||
-      show_password_reset_partial? ||
-      show_gpo_partial?
+      show_password_reset_partial?
+  end
+
+  def show_unverified?
+    show_password_reset_partial? &&
+      (!show_ipp_partial? || !show_gpo_partial?)
+  end
+
+  def service_provider_or_app_name
+    if user.identities.count == 0
+      APP_NAME
+    else
+      user.identities.last.friendly_name
+    end
+  end
```

Comment:
ref https://github.com/18F/identity-idp/pull/10823#discussion_r1647958244
Possibly, based on my slightly brittle assumption. 
If it's the user's last, most-recently added SP identity then yes but there is a way they could auth at one SP, auth at a second, then go back and do identity verification on the first one. I'm not sure how likely that is though. 🤔

Reply:
I think the way I'd expected us to implement this was using [`Profile#initiating_service_provider_issuer`](https://github.com/18F/identity-idp/blob/59c4b26602fdf22af3b788fca2a91739b0419f9b/db/schema.rb#L443). So we could look up a `ServiceProvider` by that issuer and use its `friendly_name`.